src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,294 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from lightning_utilities.core.imports import RequirementCache
+
+import pytorch_lightning as pl
+from lightning_lite.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.accelerators.cuda import CUDAAccelerator
+from pytorch_lightning.overrides.base import _LightningModuleWrapperBase, _LightningPrecisionModuleWrapperBase
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.precision import ColossalAIPrecisionPlugin, PrecisionPlugin
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+_COLOSSALAI_AVAILABLE = RequirementCache("colossalai")
+if _COLOSSALAI_AVAILABLE:
+    from colossalai.context import ParallelMode
+    from colossalai.core import global_context as gpc
+    from colossalai.gemini import ChunkManager, GeminiManager
+    from colossalai.logging import disable_existing_loggers, get_dist_logger
+    from colossalai.nn.optimizer import CPUAdam, HybridAdam
+    from colossalai.nn.parallel import ZeroDDP
+    from colossalai.tensor import ProcessGroup
+    from colossalai.utils import get_current_device
+    from colossalai.utils.model.colo_init_context import ColoInitContext
+    from colossalai.zero import ZeroOptimizer
+
+
+class ColossalAIStrategy(DDPStrategy):
```

Comment:
Important consideration: Is Colossal in anyway related to torch ddp? If not, please consider inheriting from the ParallelStrategy base class directly!

Reply:
I am fine either way, but this shouldn't block the PR since a few more strategies use DDPStrategy as the base. We can refactor it later to flatten it.

Reply:
as discussed, we will do this ref in future.